dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/AndersenAliasAnalysis.java

```diff
@@ -45,18 +42,20 @@ public class AndersenAliasAnalysis implements AliasAnalysis {
     private final Queue<Object> variables = new ArrayDeque<>();
     ///Super set of all pointer sets in this analysis
     private final ImmutableSet<Location> maxAddressSet;
-    private final Graph graph = new Graph();
-
+    private final Map<Object, Set<Object>> edges = new HashMap<>();
+    private final Map<Object, Set<Location>> addresses = new HashMap<>();
+    private final Map<Register, Set<MemEvent>> events = new HashMap<>();
+    private final Map<Register, Set<Location>> targets = new HashMap<>();
     private final Map<MemEvent, ImmutableSet<Location>> eventAddressSpaceMap = new HashMap<>();
 
     // ================================ Construction ================================
 
     private AndersenAliasAnalysis(Program program) {
```

Comment:
(NOTE: This comment refers to the whole class)

Is this just a reimplementation of the `AndersenAliasAnalysis` to be more similar to the field sensitive one?
Or does this rework involve any relevant changes (e.g., semantic change or performance improvement)